pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -6,44 +6,127 @@ import (
 	"reflect"
 	"time"
 
-	inventory "github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/inventory"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/meta"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/openshift"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/builder"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
 
 	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/logs"
+	kogitocli "github.com/kiegroup/kogito-cloud-operator/pkg/client"
 	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/shared"
 	"github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitoapp/status"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/logger"
 	oappsv1 "github.com/openshift/api/apps/v1"
 	obuildv1 "github.com/openshift/api/build/v1"
+	oimagev1 "github.com/openshift/api/image/v1"
 	oroutev1 "github.com/openshift/api/route/v1"
+	routev1 "github.com/openshift/api/route/v1"
 	buildv1 "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
 	imagev1 "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	cachev1 "sigs.k8s.io/controller-runtime/pkg/cache"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = logs.GetLogger("controller_kogitoapp")
+var log = logger.GetLogger("controller_kogitoapp")
+
+// Add creates a new KogitoApp Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	imageClient, err := imagev1.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Errorf("Error getting image client: %v", err)
+		return &ReconcileKogitoApp{}
+	}
+	imageClientInterface := imagev1.ImageV1Interface(imageClient)
+
+	buildClient, err := buildv1.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		log.Errorf("Error getting build client: %v", err)
+		return &ReconcileKogitoApp{}
+	}
+
+	buildClientInterface := buildv1.BuildV1Interface(buildClient)
```

Comment:
This is unnecessary, and confusing. Just use buildClient directly on line 69.

Reply:
LOL. I completely forgot to remove it. Thanks!